src/plan_bot.py

```diff
@@ -69,35 +69,51 @@ def process_post(
     if post.id not in post_ids_replied_to:
         # Do a case insensitive search
         if re.search("!warrenplanbot|/u/WarrenPlanBot", post.text, re.IGNORECASE):
-            # Initialize match_confidence and match_id before fuzzy searching
-            match_confidence = 0
+            # Initialize minimum match_confidence to 50% and match_id before fuzzy searching
+            match_confidence = 50
             match_id = 0
+            match_topic=""
 
             # Search topic keywords and response body for best match
             for plan in plans_dict["plans"]:
-                plan_match_confidence = fuzz.WRatio(post.text, plan["topic"])
+                plan_match_confidence = fuzz.token_sort_ratio(post.text, plan["topic"])
 
                 if plan_match_confidence > match_confidence:
                     # Set new match ID
                     match_confidence = plan_match_confidence
                     match_id = plan["id"]
+                    match_topic = plan["topic"]
                     print("new topic match: ", plan["topic"])
 
-            # Select entry from plans_dict using best match ID
-            plan_record = next(
-                plan for plan in plans_dict["plans"] if plan["id"] == match_id
-            )
-
-            reply_string = build_response_text(plan_record, post)
-
-            did_reply = reply(post, reply_string, send=send, simulate=simulate)
+            # If new topic matched with confidence > 50% select and build reply from new match_id
+            if match_id != 0:
```

Comment:
Nit: Could be simply `if match_id:`

Reply:
Agree, though I tend to use explicit if conditions instead of truthy falsey to avoid potential bugs